Prototype for the activity feed

jcolladokuri opened the pull request:
Frist couple changes to create an activity feed page, still trying to get the hang of JSP files so not sure if I implemented that correctly, any suggestions?

Commit:
Prototype for the activity feed

Frist couple changes to create an activity feed page, still trying to get the hang of JSP files so not sure if I implemented that correctly, any suggestions?

dgalbraith33 commented on src/main/java/codeu/controller/ActivityFeedServlet.java:
```diff
@@ -0,0 +1,49 @@
+
+			request.getRequestDispatcher("/WEB-INF/view/activity.jsp").forward(request, response);
+	}
+}
```
nit: Add a newline at the end of the file

Rafer45 replied:
Each tab should be two spaces, instead of a tab character.

jcolladokuri replied:
Done!

dgalbraith33 commented on src/main/webapp/WEB-INF/view/activityfeed.jsp:
```diff
@@ -0,0 +1,24 @@
+--%>
+<%-- still trying to get the hang of jsp, not sure if this is correctly implemented --%>
+<%  List<String> friends_usernames = Arrays.aslist('Ricardo', 'Elona', 'Drew');
+    String username = (String) request.getSession().getAtrribute('user');
```
You should pass this information to the JSP file via the request or response object in the servlet. The more logic we keep in the Servlet, the more we can unit test and the more errors the compiler can catch.

Check the ChatServlet to see how it passes the conversation and messages to JSP :)

dgalbraith33 commented on src/main/webapp/WEB-INF/view/activityfeed.jsp:
```diff
@@ -0,0 +1,24 @@
+%>
+    <p>There is no activity to display</p>
+<%  }
+%> 
```
nit: Add a newline here

Rafer45 commented on src/main/java/codeu/controller/ActivityServletTest.java:
```diff
@@ -0,0 +1,70 @@
+
+  @Test
+  public void testDoGet_friendUsername() throws IOException, ServletException {
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("Welcome to the activity feed page");
```
The "user" attribute usually contains the username of the client, not strings with spaces like the one used here.
Is this intentional?

jcolladokuri replied:
I changed it! Thanks for the catch

Rafer45 commented on src/main/java/codeu/controller/ActivityServletTest.java:
```diff
@@ -0,0 +1,70 @@
+    // Verify user still is redirected to the activity page but correct message is displayed
+    Mockito.verify(mockRequestDispatcher, Mockito.never()).forward(mockRequest, mockResponse);
+  }
+}
```
I think there should be a newline here.

jcolladokuri replied:
Done!

eselenic commented on src/main/webapp/WEB-INF/view/activityfeed.jsp:
```diff
@@ -0,0 +1,24 @@
+<%  List<String> friends_usernames = Arrays.aslist('Ricardo', 'Elona', 'Drew');
+    String username = (String) request.getSession().getAtrribute('user');
+
+    if(friends_usernams.contains(username)){
```
Small typo! "friends_usernams" should be "friends_usernames"

Rafer45 committed:
Fix broken type signature. My bad!

Commit:
Changes requested to first prototype of Activity Feed Page

Commit:
few bug fixes

Rafer45 commented on src/main/java/codeu/controller/ActivityFeedServlet.java:
```diff
@@ -0,0 +1,57 @@
+			request.getRequestDispatcher("/WEB-INF/view/activityfeed.jsp").forward(request, response);
+			}
+		}
+}
```
Fix the indentation on this file. Indentation on all other files looks good!

Commit:
fixed indentation

Commit:
Merge remote-tracking branch 'origin/optimization' into Jocelyn

Rafer45 committed:
Tweak indentation

Rafer45 committed:
Tweak whitespace

Commit:
added username to codeowners

Commit:
Merge branch 'Prototype-for-activity-feed-page' of https://github.com/CodeUCloudJockeys/CodeU-Spring-2018 into Jocelyn

Rafer45 commented:
LGTM!

Commit:
Handles adding friends

Activity related to friends is sill on the works as well as some test cases. Did not make the friend profile too ellaborated for now.

Rafer45 commented on src/main/java/codeu/model/data/Friend.java:
```diff
@@ -0,0 +1,24 @@
+import java.time.Instant;
+import java.util.UUID;
+
+public class Friend{
```
Make the indentation two spaces, instead of tab characters.

Rafer45 commented on src/main/java/codeu/model/data/Friend.java:
```diff
@@ -0,0 +1,24 @@
+	public String getName(){
+		return name;
+	}
+}
```
Add a newline here

Rafer45 commented on src/main/webapp/WEB-INF/web.xml:
```diff
@@ -67,6 +67,16 @@
   </servlet-mapping>
 
   <servlet>
+<<<<<<< HEAD
```
Remove this line; it's from a merge conflict.

Rafer45 commented on src/main/webapp/WEB-INF/web.xml:
```diff
@@ -67,6 +67,16 @@
++    <servlet-name>ActivityFeedServlet</servlet-name>
++    <url-pattern>/activity</url-pattern>
++  </servlet-mapping>
+=======
```
Same for this line.

Rafer45 commented on src/main/webapp/WEB-INF/web.xml:
```diff
@@ -75,5 +85,6 @@
     <servlet-name>AdminServlet</servlet-name>
     <url-pattern>/admin</url-pattern>
   </servlet-mapping>
+>>>>>>> origin/optimization
```
Same for this line.

Rafer45 reviewed (changes requested):
Just some whitespace tweaks and removing git merge conflict lines
